src/server.js

```diff
@@ -33,21 +37,42 @@ app.get("/presentations/:id", (req, res) => {
     (presentation) => presentation._id == presID,
   );
   if (presentation == null) {
-    res.render('404', {message:"Presentation not found"})
-    return
+    res.render("404", { message: "Presentation not found" });
+    return;
   }
+  const favorited = favorites.includes(presentation._id);
   res.render("presentation_description", {
-    presentation: presentation,
+    presentation: { ...presentation, favorited },
   });
 });
 
+// temporary to hold favorite state in server state until we get viewstate in place
+let favorites = [];
+
 app.post("/search", (req, res) => {
   const presentationsFound = searchPresentations(
     req.body.search,
     presentations,
   );
 
-  res.render("search", { presentations: presentationsFound , layout: false});
+  res.render("search", { presentations: presentationsFound, layout: false });
+});
+
+app.put("/presentations/favorite/:id", (req, res) => {
+  const id = req.params.id;
+  // what if id doesn't exist???
+
+  // we'll toggle favorite on and off with this endpoint
+  const isExistingFavorite = favorites.includes(id);
+  favorites = isExistingFavorite
+    ? favorites.filter((f) => f !== id)
+    : [...favorites, id];
```

Comment:
Might be better to use a Set instead?

Reply:
Yea, probably. This local array gets replaced in the follow-up PR so I might do a refactor there.